server/aws/client_cache.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package aws
+
+import (
+	sdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+
+	"github.com/pkg/errors"
+)
+
+// DefaultRegion describes default region in aws
+const DefaultRegion = "us-east-2"
+
+type clientCache struct {
+	store CredentialStoreService
+	cache map[string]*Client // appID - Client
+}
+
+func newClientCache(store CredentialStoreService) (*clientCache, error) {
```

Comment:
How does the cache gets updated / invalidated? For example, when an app gets added/deleted. Not sure if the apps will change the credentials at any moment.